src/rpc/rawtransaction.cpp

```diff
@@ -1262,6 +1263,552 @@ static UniValue testmempoolaccept(const JSONRPCRequest& request)
+            "            \"hex\":\"hex\",         (string) Hex encoded public key\n"
+            "            \"type\":\"type\",       (string) The output type\n"
+            "            \"reqSigs\": n,        (numeric) The required signatures\n"
+            "            \"addresses\": [       (json array of string)\n"
+            "              \"address\"           (string) Bitcoin address\n"
+            "               ,...\n"
+            "            ],\n"
+            "        }\n"
+            "      \"witness_script\" : {       (json object, optional)\n"
+            "          \"script\" : {            (json object)\n"
+            "            \"asm\" : \"asm\",        (string) The asm of the script\n"
+            "            \"hex\" : \"hex\",        (string) The hex of the script\n"
+            "        }\n"
+            "      }\n"
+            "      \"bip32_derivs\" : [          (array of json objects, optional)\n"
+            "        {\n"
+            "          \"pubkey\" : \"pubkey\",                     (string) The public key this path corresponds to\n"
+            "          \"master_fingerprint\" : \"fingerprint\"     (string) The fingerprint of the master key\n"
+            "          \"path\" : \"path\",                         (string) The path\n"
+            "          }\n"
+            "        }\n"
+            "        ,...\n"
+            "      ],\n"
+            "      \"unknown\" : {                (json object) The unknown global fields\n"
+            "        \"key\" : \"value\"            (key-value pair) An unknown key-value pair\n"
+            "         ...\n"
+            "      },\n"
+            "    }\n"
+            "    ,...\n"
+            "  ]\n"
+            "}\n"
+
+            "\nExamples:\n"
+            + HelpExampleCli("decodepsbt", "\"psbt\"")
+    );
+
+    RPCTypeCheck(request.params, {UniValue::VSTR});
+
+    // Unserialize the transactions
+    PartiallySignedTransaction psbtx;
+    std::string error;
+    if (!DecodePSBT(psbtx, request.params[0].get_str(), error)) {
+        throw JSONRPCError(RPC_DESERIALIZATION_ERROR, strprintf("TX decode failed %s", error));
+    }
+
+    UniValue result(UniValue::VOBJ);
+
+    // Add the decoded tx
+    UniValue tx_univ(UniValue::VOBJ);
+    TxToUniv(CTransaction(*psbtx.tx), uint256(), tx_univ, false);
+    result.pushKV("tx", tx_univ);
+
+    // Unknown data
+    UniValue unknowns(UniValue::VOBJ);
+    for (auto entry : psbtx.unknown) {
+        unknowns.pushKV(HexStr(entry.first), HexStr(entry.second));
+    }
+    result.pushKV("unknown", unknowns);
+
+    // inputs
+    UniValue inputs(UniValue::VARR);
+    for (const PSBTInput& input : psbtx.inputs) {
+        UniValue in(UniValue::VOBJ);
+        // UTXOs
+        if (!input.witness_utxo.IsNull()) {
+            const CTxOut& txout = input.witness_utxo;
+
+            UniValue out(UniValue::VOBJ);
+
+            out.pushKV("value", ValueFromAmount(txout.nValue));
```

Comment:
For compatibility with `listunspent` you may want to call this `"amount"`.

Reply:
Done